tests/integration/nm/linux_bridge_test.py

```diff
@@ -25,19 +25,62 @@
 from libnmstate import schema
 from libnmstate.schema import LinuxBridge as LB
 
-from .testlib import mainloop
+from ..testlib import iproutelib
+from .testlib import mainloop_run
 
 
 BRIDGE0 = 'brtest0'
 
 
 @pytest.fixture
-def bridge_minimum_config():
-    return {LB.CONFIG_SUBTREE: {LB.PORT_SUBTREE: []}}
+def bridge0_with_port0(port0_up):
+    port_name = port0_up[schema.Interface.KEY][0][schema.Interface.NAME]
+    bridge_desired_state = _create_bridge_config((port_name,))
+    with _bridge_interface(bridge_desired_state):
+        yield _get_bridge_current_state()
 
 
-@pytest.fixture
-def bridge_default_config():
+def test_create_and_remove_minimum_config_bridge():
+    bridge_desired_state = {LB.CONFIG_SUBTREE: {LB.PORT_SUBTREE: []}}
+
+    with _bridge_interface(bridge_desired_state):
+
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_current_state
+        assert not bridge_current_state[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE]
+
+    assert not _get_bridge_current_state()
+
+
+def test_create_and_remove_bridge(port0_up):
+    port_name = port0_up[schema.Interface.KEY][0][schema.Interface.NAME]
+    bridge_desired_state = _create_bridge_config((port_name,))
+    with _bridge_interface(bridge_desired_state):
+
+        bridge_current_state = _get_bridge_current_state()
+        assert bridge_desired_state == bridge_current_state
+
+    assert not _get_bridge_current_state()
+
+
+@iproutelib.ip_monitor_assert_stable_link_up(BRIDGE0)
+def test_add_port_to_existing_bridge(bridge0_with_port0, port1_up):
+    port_name = port1_up[schema.Interface.KEY][0][schema.Interface.NAME]
+    _add_ports_to_bridge_config(bridge0_with_port0, (port_name,))
+
+    _modify_bridge(bridge0_with_port0)
+
+    assert bridge0_with_port0 == _get_bridge_current_state()
+
+
+def _add_ports_to_bridge_config(bridge_state, ports):
+    ports_config = _create_bridge_ports_config(ports)
+    bridge_state[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] += ports_config
+
+
+def _create_bridge_config(ports):
+
```

Comment:
please remove this empty line (or is there some deeper meaning to it 0_o)

Reply:
Done